packages/tools/client-debugger/client-debugger-view/src/components/DataObjectsView.tsx

```diff
@@ -18,10 +32,61 @@ export type DataObjectsViewProps = HasContainerId;
  * Dispatches data object rendering based on those provided view {@link DataObjectsViewProps.renderOptions}.
  */
 export function DataObjectsView(props: DataObjectsViewProps): React.ReactElement {
+	const { containerId } = props;
+
+	const messageRelay = useMessageRelay();
+
+	const [rootDataHandles, setRootDataHandles] = React.useState<
+		Record<string, RootHandleNode> | undefined
+	>();
+
+	React.useEffect(() => {
+		const inboundMessageHandlers: InboundHandlers = {
+			["ROOT_DATA_VISUALIZATIONS"]: (untypedMessage) => {
+				const message: RootDataVisualizationsMessage =
+					untypedMessage as RootDataVisualizationsMessage;
+
+				if (message.data.containerId === containerId) {
+					setRootDataHandles(message.data.visualizations);
+
+					return true;
+				} else {
+					return false;
+				}
+			},
+		};
+
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		messageRelay.postMessage({
+			type: "GET_ROOT_DATA_VISUALIZATIONS",
+			data: {
+				containerId,
+			},
+		});
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [containerId, rootDataHandles, messageRelay]);
```

Comment:
You'll want to omit `rootDataHandles` here, otherwise we will rerun the effect (including sending a new request for all of the data, which will likely result in an infinite async loop of data messaging). But we probably do want to include `setRootDataHandles` for completeness, since we call it in the effect (I don't believe React will ever give us a new setter function, but just in case).